# Arcseconds and parsec

In [88]:
%pylab ipympl

Populating the interactive namespace from numpy and matplotlib


In [89]:
from astropy import units
from astropy import constants

By default the sin function takes the angle in units of radians where a complete turn of $2\pi \mathrm{rad}$ corresponds to $360^{\circ}$. For example, $90^{\circ}$ corresponds to $\frac{1}{2}\pi \mathrm{rad}$:

In [90]:
sin(0.5*pi)

1.0

Another measure for angle are minutes and seconds. 60 minutes are one degree.

In [22]:
a=90*60*units.arcmin

In [23]:
sin(a)

<Quantity 1.>

In [25]:
a=90*3600*units.arcsec

In [26]:
sin(a)

<Quantity 1.>

One arcsec is very small, and the sin of one arcsec is very small:

In [92]:
sin(1*units.arcsec)

<Quantity 4.84813681e-06>

sin(45 deg):

In [29]:
sin(pi/4.)

0.7071067811865475

In [31]:
1/sqrt(2)

0.7071067811865475

Convert units conveniently with the unit package, for example one arcmin in rad:

In [93]:
units.arcmin.to('rad')

0.0002908882086657216

In order to make a plot we create an array containing numbers with units:

In [94]:
a = linspace(0,90*60*units.arcmin,100)

The 10$^\mathrm{th}$ element of the angle array $a$:

In [96]:
a[10]

<Quantity 545.45454545 arcmin>

Note how the unit of angle is changed when plotting the second line:

In [97]:
close(1);figure(1)
plot(a,sin(a),label="$ \sin (\\alpha)$")
plot(a,a.to('rad'),'--',label="$ \\alpha \mathrm{[rad]}$")
xlabel("$\\alpha [\mathrm{'}]$")
legend()

Canvas(toolbar=Toolbar(toolitems=[('Home', 'Reset original view', 'home', 'home'), ('Back', 'Back to previous …

The key point is that for small $\alpha$ (certainly a couple hundred arcmin) $\sin(\alpha) \approx \alpha$ is a very good approximation. 

## Astronomical unit



In [77]:
constants.au

<<class 'astropy.constants.iau2015.IAU2015'> name='Astronomical Unit' value=149597870700.0 uncertainty=0.0 unit='m' reference='IAU 2012 Resolution B2'>

## Parsec
The distance $D$ of an object from which one astronomical unit $AU$ is seen at an angle 1 arcsec. Because for such a small angle $\sin(\alpha) \approx \alpha$ we have
for $\alpha = 1\mathrm{'} = 4.848\times10^{-6}\mathrm{rad}$ 
$$
\alpha \approx \sin \alpha = \frac{AU}{D}
$$
where we just need to remember that that because $\alpha$ as an argument of $\sin$ must be in $\mathrm{rad}$, $\alpha$ on the far left of this equation must also be in radians.

In [98]:
# one arcsec in rad:
units.arcsec.to('rad')

4.84813681109536e-06

In [99]:
# Then one parsec is
D = constants.au / units.arcsec.to('rad')
D

<Quantity 3.08567758e+16 m>

In [100]:
# parsec in AU:
D.to('AU')

<Quantity 206264.8062471 AU>

In [101]:
# parsec in lightyear:
D.to('lightyear')

<Quantity 3.26156378 lyr>